src/main/scala/cromwell/binding/Scatter.scala

```diff
@@ -14,14 +14,59 @@ object Scatter {
 
 /**
  * Scatter class.
- * @param index Index of the scatter block. The index is computed during tree generation to reflect wdl scatter blocks structure.
+ * @param namespaceIndex Index of the scatter block. The index is computed during tree generation to reflect wdl scatter blocks structure.
  * @param item Item which this block is scattering over
  * @param collection Wdl Expression corresponding to the collection this scatter is looping through
+ * @param parent Parent of this scatter
+ * @param scatterIndex When the parent is a scatter, the index within the parent scatter.
  */
-case class Scatter(index: Int, item: String, collection: WdlExpression, parent: Option[Scope]) extends Scope {
+case class Scatter(namespaceIndex: Int,
+                   item: String,
+                   collection: WdlExpression,
+                   override val parent: Option[Scope],
+                   override val scatterIndex: Option[Int] = None) extends ParentScope with ChildScope {
```

Comment:
`scatterIndex` strikes me as another thing that shouldn't leak down to the namespace layer.  Excluding this PR, we have no information about execution in the `cromwell.binding` layer, and hopefully we can keep it that way!